weather_app/greeting.py

```diff
@@ -1,94 +1,181 @@
-    Plot line graph of 24hr forecast, and bar graph of 10-day max & min.
+    """Get graphs of air temperature forecasts.
+
+    Parameters
+    ----------
+    data: pd.Series
+        Temperature values with a timezone-aware date-time index.
+
+    Returns
+    -------
+    A line graph of the 24hr forecast, and a bar graph of the 10-day max & min
+    temperature forecast, stored in text buffers.
     """
     temp24H = data[:24]
-    temp24H_graph = StringIO()
     fig = px.line(y=temp24H, x=temp24H.index.astype(str),
-                  title="24 Hour Forecast", width=700, height=500)
-    fig.update_xaxes(title_text='Time')
-    fig.update_yaxes(title_text='Air temperature in deg C')
-
-    fig.write_html(temp24H_graph, include_plotlyjs='cdn',
-                   full_html=False)
+                  title="24 Hour Forecast")
+    # Rename axes labels and disable zoom
+    fig.update_xaxes(title_text='Time', fixedrange=True)
+    fig.update_yaxes(title_text='Air temperature in deg C', fixedrange=True)
+    fig.update_layout(paper_bgcolor='azure', plot_bgcolor='azure')
+    # Write graph to text buffer
+    temp24H_graph = StringIO()
+    fig.write_html(temp24H_graph, include_plotlyjs='cdn', full_html=False)
 
     temp10D = data.resample('1D').agg(['max', 'min'])
-    temp10D_graph = StringIO()
     fig2 = px.bar(temp10D, color_discrete_sequence=['orangered', 'cyan'],
-                  barmode='group', title="10 Day Forecast", width=700,
-                  height=500)
-    fig2.update_xaxes(title_text='Day')
-    fig2.update_yaxes(title_text='Air temperature in deg C')
-    fig2.write_html(temp10D_graph, include_plotlyjs='cdn',
-                    full_html=False)
+                  barmode='group', title="10 Day Forecast")
+    # Rename axes labels and disable zoom
+    fig2.update_xaxes(title_text='Day', fixedrange=True)
+    fig2.update_yaxes(title_text='Air temperature in deg C', fixedrange=True)
+    fig2.update_layout(paper_bgcolor='azure', plot_bgcolor='azure')
+    # Write graph to text buffer
+    temp10D_graph = StringIO()
+    fig2.write_html(temp10D_graph, include_plotlyjs='cdn', full_html=False)
     return temp24H_graph, temp10D_graph
 
 
+def get_wikipedia_info(*location):
+    """Get brief details about a location from Wikipedia.
+
+    Parameters
+    ----------
+    location: str
+        Location name(s) to search for.
+
+    Returns
+    -------
+    Summary text about the location.
+    """
+    # Search for articles about location(s), and select first result
+    wiki_query = wikipedia.search(f"{location}")[0]
+
+    return wikipedia.summary(wiki_query, sentences=5)
+
+
+def draw_map(lat, lon):
+    """Draw the map of a location given latitude and longitude.
+
+    Parameters
+    ----------
+    lat, lon: int or float
+        Latitude and longitude values, respectively.
+
+    Returns
+    -------
+    A string with HTML code for a map.
+    """
+    return folium.Map(location=(lat, lon), zoom_start=14)._repr_html_()
```

Comment:
I don't know Folium that well but is there a different way instead of calling `_repr_html`? I bring this up since methods that start with an `_` are consider "private" and we may want to avoid directly calling them?

Reply:
In the [Using folium, with flask][1] docs, the `_repr_html_` method is the one used, but there's also a 'public' `save` method. I compared the two, and found that:

- The `_repr_html_` method returns a `div` element with an `iframe`, which is much more suited for plugging into HTML templates.
- The `save` method returns a more stand-alone file, with a `<!DOCTYPE html>` tag and all.

[1]: https://python-visualization.github.io/folium/flask.html

Reply:
Thanks for looking into this.